test/hotspot/jtreg/compiler/integerArithmetic/TestNegMultiply.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8273454
+ * @summary Test transformation (-a)*(-b) = a*b
+ *
+ * @run main/othervm -XX:-TieredCompilation -XX:-BackgroundCompilation -XX:-UseOnStackReplacement TestNegMultiply
+ *
+ */
+
+import java.util.Random;
+
+public class TestNegMultiply {
+    private static Random random = new Random();
+    private static final int TEST_COUNT = 2000;
+
+    private static int test(int a, int b) {
+        return (-a) * (-b);
+    }
+
+    private static void testInt(int a, int b) {
+        int expected = (-a) * (-b);
+        for (int i = 0; i < 20_000; i++) {
+            if (expected != test(a, b)) {
+                throw new RuntimeException("Incorrect result.");
+            }
+        }
+    }
+
+    private static long test(long a, long b) {
+        return (-a) * (-b);
+    }
+
+    private static void testLong(long a, long b) {
+        long expected = (-a) * (-b);
+        for (int i = 0; i < 20_000; i++) {
+            if (expected != test(a, b)) {
+                throw new RuntimeException("Incorrect result.");
+            }
+        }
+    }
```

Comment:
How about calculating the expected value outside the iteration to avoid it to be compiled？

``` java   
 private static void testLong() {
        for (int i = 0; i < 20_000; i++) {
            long a = random.nextLong();
            long b = random.nextLong();
            long expected = (-a) * (-b);
            if (expected != test(a, b)) {
                throw new RuntimeException("Incorrect result.");
            }
        }
    }
```